src/grid/rtransform.py

```diff
@@ -77,6 +87,15 @@ def transform_1d_grid(self, oned_grid):
         """
         if not isinstance(oned_grid, OneDGrid):
             raise TypeError(f"Input grid is not OneDGrid, got {type(oned_grid)}")
+        # check domain
+        if (
+            oned_grid.domain[0] < self.tf_domain[0]
+            or oned_grid.domain[1] > self.tf_domain[1]
+        ):
+            raise ValueError(
+                "Input grid domain do not match with tf domain.\n"
```

Comment:
To be more clear for the user, we can say:
"Given 1D grid domain does not match the transformation domain"

Reply:
I also prefer `domain` and `codomain`. Just not sure would it bring ambiguity with the `domain` in grid.